src/plugins/navigation/public/plugin.tsx

```diff
@@ -129,39 +127,65 @@ export class NavigationPublicPlugin
 
     const config = this.initializerContext.config.get();
     const {
-      solutionNavigation: { featureOn: isSolutionNavigationFeatureOn },
+      solutionNavigation: { defaultSolution },
     } = config;
 
     const onCloud = cloud !== undefined; // The new side nav will initially only be available to cloud users
     const isServerless = this.initializerContext.env.packageInfo.buildFlavor === 'serverless';
-    const isSolutionNavEnabled = isSolutionNavigationFeatureOn && onCloud && !isServerless;
-    this.isSolutionNavEnabled$ = of(isSolutionNavEnabled);
 
-    if (isSolutionNavEnabled) {
-      chrome.project.setCloudUrls(cloud);
-      this.addDefaultSolutionNavigation({ chrome });
+    let isFeatureEnabled$ = of(false);
+    this.isSolutionNavEnabled$ = of(false);
 
-      this.isSolutionNavEnabled$ = combineLatest([
-        core.settings.globalClient.get$<boolean>(ENABLE_SOLUTION_NAV_UI_SETTING_ID),
-        core.settings.globalClient.get$<SolutionNavigationOptInStatus>(
-          OPT_IN_STATUS_SOLUTION_NAV_UI_SETTING_ID
-        ),
-        this.userProfileOptOut$,
-      ]).pipe(
-        takeUntil(this.stop$),
-        debounceTime(10),
-        map(([enabled, status, userOptedOut]) => {
-          if (!enabled || userOptedOut === true) return false;
-          if (status === 'hidden' && userOptedOut === undefined) return false;
-          return true;
-        })
+    if (cloudExperiments) {
+      isFeatureEnabled$ = from(
+        cloudExperiments.getVariation(SOLUTION_NAV_FEATURE_FLAG_NAME, false)
+      ).pipe(
+        shareReplay(1),
+        map((isFeatureFlagEnabled) => isFeatureFlagEnabled && onCloud && !isServerless)
```

Comment:
Looks like we don't even need to call `cloudExperiments.getVariation` if onCloud && !isServerless . can this check be moved higher?

Reply:
Maybe even `isFeatureEnabled$` is not needed if not on cloud or serverless.

Reply:
Good point, addressed in https://github.com/elastic/kibana/pull/179913/commits/cc93dd7af1d7ef0bbf78656e2459157acdc4b65e

Reply:
> Maybe even isFeatureEnabled$ and this.isSolutionNavEnabled$ are not needed if not on cloud.

Yeah can probably be refactored one step further 👍 I'll follow up on that.